src/pathfinding_igvc/test/path-finding-test.cpp

```diff
@@ -0,0 +1,32 @@
+#include <PathFinding.h>
+#include <gtest/gtest.h>
+
+
+TEST(PathFinding, testWeightedSum1){
+
+    //PathFinding test_node(NULL,NULL,NULL);
+
+    std::vector<float> testVec;
+    testVec.push_back(3);
+    testVec.push_back(3);
+    testVec.push_back(3);
+
+    EXPECT_EQ(6.0, PathFinding::weightedXSum(testVec, 3));
+}
+
+TEST(PathFinding, testWeightedSum2){
+
+    //PathFinding test_node(NULL,NULL,NULL);
+
+    std::vector<float> testVec;
+    testVec.push_back(1);
+    testVec.push_back(2);
+    testVec.push_back(3);
+
+    EXPECT_EQ(1.0, PathFinding::weightedYSum(testVec, 1));
+}
+
```

Comment:
Once we have the function I mentioned above, we can try to test:
- different starting orientation (what if we're perpendicular to the first vector in the path? what if we're facing the opposite direction?)
- different starting position (what if we're a meter away from the start of the path? we should make sure to have a few tests located far away from (0,0) to make sure that doesn't break anything)
- paths with immediate sharp left/right turns
- paths with a few consecutive vectors that point in radically different directions (ie. directly right, directly left, 45 degrees right, etc.)